SystemInformer/procprv.c

```diff
@@ -231,73 +231,108 @@ PH_CIRCULAR_BUFFER_ULONG64 PhMaxIoWriteHistory;
 static PPH_HASHTABLE PhpSidFullNameCacheHashtable = NULL;
 
 BOOLEAN PhProcessProviderInitialization(
-    VOID
+    _In_ HWND hwnd
```

Comment:
Passing a window handle to PhProcessProviderInitialization breaks the abstraction between the UI layer and background provider and adds lots of unnecessary checks. 

You should move PhProcessImageListInitialization out of PhProcessImageListInitialization, calling it right before PhProcessProviderInitialization and then remove the window handle from PhProcessProviderInitialization and only pass the handle to PhProcessImageListInitialization.

Reply:
Here i did not understand where is problem